lib/redlock/client.rb

```diff
@@ -228,6 +281,36 @@ def lock_instances(resource, ttl, options)
       end
     end
 
+    def try_get_remaining_ttl(resource)
```

Comment:
@maltoe ultimately, this has introduced more complexity than anticipated, but it should cover all the bases. Let me know what you think.